ademiltonnunes/marvel-platform
# Refactor onboarding

# Description
This PR refactors the layout provision among pages and components, streamlines the onboarding process, and implements a slide transition for the onboarding screen.

Key changes and implementation details:

1. Layout Refactoring:
   - Implemented a new `Layout` component that serves as a wrapper for all pages
   - Moved common layout elements (e.g., header, footer, sidebar) to the `Layout` component
   - Updated page components to use the new `Layout` wrapper, reducing code duplication

2. Single-page Onboarding Process:
   - Created a new `OnboardingPage` component that handles the entire onboarding flow
   - Implemented a step-based system within `OnboardingPage` to manage different stages of onboarding
   - Used React state to track progress and display appropriate content for each step

3. HOC Modification for Redirection:
   - Updated the existing Higher Order Component (HOC) for authentication
   - Added a check in the HOC to verify if the user has completed onboarding
   - If onboarding is incomplete, the HOC now redirects to the `OnboardingPage` instead of the requested protected page
   - Implemented this using React Router's `Navigate` component for seamless redirection

4. Slide Transition for Onboarding:
   - Utilized React Transition Group for managing enter/exit transitions
   - Created a custom `SlideTransition` component that wraps onboarding content
   - Implemented CSS transitions for smooth sliding effects between onboarding steps
   - Added configurable direction prop (left/right) to `SlideTransition` for flexible animations

These changes improve code organization, enhance user experience during onboarding, and ensure proper redirection for new users. The refactored layout system provides better consistency across the application, while the streamlined onboarding process with animated transitions offers a more engaging user experience.

## Type of change

# How Has This Been Tested?
- [x] Tested layout changes across various pages and components to ensure consistency
- [x] Verified the new single-page onboarding process flow
- [x] Checked HOC redirection for users who haven't completed onboarding
- [x] Tested slide transition animation on the onboarding screen
- [x] Verified proper functionality on different devices and screen sizes

Please test the following scenarios:
1. Navigate through different pages to ensure correct layout application
2. Go through the entire onboarding process as a new user
3. Attempt to access protected pages before completing onboarding
4. Check the smoothness of the slide transition during onboarding
5. Test responsiveness of the new layout and onboarding process on various screen sizes
